internal/pkg/aws/route53/route53.go

```diff
@@ -152,7 +152,7 @@ type filterZoneFunc func(*route53.HostedZone) bool
 func filterHostedZones(zones []*route53.HostedZone, fn filterZoneFunc) []*route53.HostedZone {
 	var hostedZones []*route53.HostedZone
 	for _, hostedZone := range zones {
-		if fn(hostedZone) {
+		if fn(hostedZone) && !isPrivateHostedZone(hostedZone.Config) {
```

Comment:
Maybe we should update this `filterHostedZones` func name too.

Reply:
An alternative!
```go
func filterHostedZones(zones []*route53.HostedZone, filters filterZoneFunc...) []*route53.HostedZone {
  for _, hostedZone := range zones {
     match := true
     for _, fn := range filters {
         if !fn(hostedZone) {
            match = false
            break
         }
     }
  }
}


hostedZones := filterHostedZones(resp.HostedZones, matchesDomain(domainName), matchesPublic())
```